uefi/src/data_types/strs.rs

```diff
@@ -24,6 +25,23 @@ pub enum FromSliceWithNulError {
     NotNulTerminated,
 }
 
+impl Display for FromSliceWithNulError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::InvalidChar(_usize) => "invalid character",
```

Comment:
For variants like this that contain extra info, I think we should go ahead and include that in the error message: `write!(f, "invalid character at index {index}")`